src/test/libcephfs/snapdiff.cc

```diff
@@ -0,0 +1,1831 @@
+  //
+  ASSERT_EQ(-EINVAL, test_mount.readfull_and_compare(snapdiff_path, "fileA",
+   "hello world AGAIN"));
+
+  // fileB to get proper content from snap2
+  ASSERT_EQ(0, test_mount.readfull_and_compare(snapdiff_path, "fileB", "hello world again in B"));
+
+  // fileC to be marked as removed and it has got content from snap1
+  ASSERT_EQ(0, test_mount.readfull_and_compare(snapdiff_path, "~fileC", "hello world in another file"));
+
+  std::cout << "------------- closing -------------" << std::endl;
+
+  ASSERT_EQ(0, test_mount.unlink("fileA"));
+  ASSERT_EQ(0, test_mount.unlink("fileB"));
+  ASSERT_EQ(0, test_mount.unlink("fileD"));
+  ASSERT_EQ(0, test_mount.rmsnap("snap1"));
+  ASSERT_EQ(0, test_mount.rmsnap("snap2"));
+}
+
+/* The following method creates some files/folders/snapshots layout,
+   described in the sheet below.
+   We're to test SnapDiff against that structure.
+
+* where:
+  - xN denotes file 'x' version N.
+  - X denotes folder name
+  - * denotes no/removed file/folder
+
+#     snap1        snap2      snap3      head
+# a1     |     a1     |    a3    |    a4
+# b1     |     b2     |    b3    |    b3
+# c1     |     *      |    *     |    *
+# *      |     d2     |    d3    |    d3
+# f1     |     f2     |    *     |    *
+# ff1    |     ff1    |    *     |    *
+# g1     |     *      |    g3    |    g3
+# *      |     *      |    *     |    h4
+# i1     |     i1     |    i1    |    i1
+# S      |     S      |    S     |    S
+# S/sa1  |     S/sa2  |    S/sa3 |    S/sa3
+# *      |     *      |    *     |    S/sh4
+# *      |     T      |    T     |    T
+# *      |     T/td2  |    T/td3 |    T/td3
+# C      |     *      |    *     |    *
+# C/cc1  |     *      |    *     |    *
+# C/C1   |     *      |    *     |    *
+# C/C1/c1|     *      |    *     |    *
+# G      |     *      |    G     |    G
+# G/gg1  |     *      |    G/gg3 |    G/gg3
+# *      |     k2     |    *     |    *
+# *      |     l2     |    l2    |    *
+# *      |     K      |    *     |    *
+# *      |     K/kk2  |    *     |    *
+# *      |     *      |    H     |    H
+# *      |     *      |    H/hh3 |    H/hh3
+# I      |     I      |    I     |    *
+# I/ii1  |     I/ii2  |    I/ii3 |    *
+# I/iii1 |     I/iii1 |    I/iii3|    *
+# *      |     *      |   I/iiii3|    *
+# *      |    I/J     |  I/J     |    *
+# *      |   I/J/i2   |  I/J/i3  |    *
+# *      |   I/J/j2   |  I/J/j2  |    *
+# *      |   I/J/k2   |    *     |    *
+# *      |     *      |  I/J/l3  |    *
+# L      |     L      |    L     |    L
+# L/ll1  |    L/ll1   |   L/ll3  |    L/ll3
+# L/LL   |    L/LL    |  L/LL    |    L/LL
+# *      |    L/LL/ll2|  L/LL/ll3|    L/LL/ll4
+# *      |    L/LM    |    *     |    *
+# *      |    L/LM/lm2|    *     |    *
+# *      |    L/LN    |    L/LN  |    *
+*/
+void prepareSnapDiffCases(TestMount& test_mount)
+{
+  //************ snap1 *************
+  ceph_assert(0 < test_mount.write_full("a", "file 'a' v1"));
```

Comment:
These should be using `ASSERT_EQ`. I see this is a static procedure. You could make this a method in a parent test class.